CHANGELOG.md

```diff
@@ -20,6 +20,12 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### New features
 
+* The new config option `snapshot.auto-track` lets you automatically track only
```

Comment:
I just had this possibly crazy idea: could we replace this option with a sparse pattern? Do we need separate concepts for sparse patterns and auto-tracked files?

`jj file (un)track` could modify sparse patterns for the working copy. We could rename "sparse patterns" if that's useful, much of the other discussion in this PR would still apply.

I had this thought after writing https://github.com/martinvonz/jj/pull/4338#discussion_r1741300502, this also follows some other assumptions I made about the goals of this that might be wrong.

Reply:
The main issue that I can think of with spare files is that it's possible that the file X is in the repo and outside the sparse set, and there is a different X in the working copy. I'm not sure what is best to do in this case, but complaining loudly and treating this like a conflict of sorts seems like an option.

Reply:
`jj file untrack` would remove the file from the repo and from the sparse pattern. There would also be a way (as there probably already exists) to change the sparse pattern without changing the repo, which would remove the file from the working copy.

One issue is whether we'd still try to warn people if they `jj file untrack` a file that's not `.gitignore`-d. This would be more difficult with the sparse files approach as opposed to the `snapshot.auto-track` approach.

Reply:
Another issue here is whether sparse patterns can handle (currently, or can be sensibly changed to handle) gitignored-but-still-tracked files.

Reply:
Making this function properly might be a bit trickier than I initially thought, but I think it might still be worth thinking about.

The analogue to `snapshot.auto-track = "none()"` would be to have the sparse patterns exclude any files that are not in the working copy commit. If this set was constant, that would be it; the "trickier than I initially thought" part is that currently the sparse set is the property of the working copy, while the set of files in the commit depends on the commit.

Even less conclusively, perhaps it's inevitable to have two separate configs for patters that are explicitly added to the sparse files, and patterns that are explicitly subtracted, but even then it might be useful to think in terms of these as methods to determine a single final set of files in the working copy that are tracked + snapshotted.